packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding/views/onboarding_phone_page.dart

```diff
@@ -0,0 +1,93 @@
+class OnboardingPhonePage extends StatelessWidget {
+  const OnboardingPhonePage({super.key});
+
+  @override
+  Widget build(BuildContext context) => RxForceUnfocuser(
+        child: RxUnfocuser(
+          child: Scaffold(
+            body: Padding(
+              padding: const EdgeInsets.symmetric(vertical: 10, horizontal: 48),
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                children: [
+                  Column(
+                    children: [
+                      Text(
+                        context.l10n.featureOnboarding.enterPhoneNumber,
+                        style: context.designSystem.typography.h1Reg20,
+                      ),
+                      const SizedBox(height: 48),
+                      const PhoneNumberPicker(),
+                    ],
+                  ),
+                  Column(
+                    children: [
+                      RxBlocBuilder<OnboardingPhoneBlocType, bool>(
+                        state: (bloc) => bloc.states.submitPhoneNumberEnabled,
+                        builder: (context, enabledSnapshot, bloc) =>
+                            RxBlocBuilder<OnboardingPhoneBlocType, bool>(
+                          state: (bloc) => bloc.states.isLoading,
+                          builder: (context, loadingSnapshot, bloc) {
+                            final enabled = enabledSnapshot.data ?? false;
+                            final loading = loadingSnapshot.data ?? false;
+                            return GradientFillButton(
+                              text: context.l10n.featureOnboarding.continueText,
+                              state: enabled
+                                  ? (loading
+                                      ? ButtonStateModel.loading
+                                      : ButtonStateModel.enabled)
+                                  : ButtonStateModel.disabled,
+                              onPressed: enabled && !loading
+                                  ? () => context
+                                      .read<OnboardingPhoneBlocType>()
+                                      .events
+                                      .submitPhoneNumber()
+                                  : null,
+                            );
+                          },
+                        ),
+                      ),
+                      const OnboardingErrorListener(),
+                      RxBlocListener<OnboardingPhoneBlocType, UserModel>(
+                        state: (bloc) => bloc.states.user,
+                        listener: _onPhoneSubmitted,
+                      ),
+                    ],
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      );
+
+  Future<void> _onPhoneSubmitted(BuildContext context, UserModel user) async {
+    final routerBloc = context.read<RouterBlocType>();
```

Comment:
Avoid using local variable for caching the bloc instance.

Reply:
This was done in order to fix the `Do not use BuildContext across asynchronous gaps.` warning (lint `use_build_context_synchronously` is not disabled).